subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/report/DefaultPerformanceExecutionDataProvider.java

```diff
@@ -69,7 +69,7 @@ private PerformanceReportScenario queryAndSortExecutionData(List<PerformanceTest
             .map(PerformanceTestExecutionResult::getTeamCityBuildId)
             .filter(StringUtils::isNotBlank)
             .collect(toList());
-        PerformanceTestHistory history = resultsStore.getTestResults(teamCityExecutionsOfSameScenario.get(0).getPerformanceExperiment(), DEFAULT_RETRY_COUNT, PERFORMANCE_DATE_RETRIEVE_DAYS, ResultsStoreHelper.determineChannel(), teamcityBuildIds);
+        PerformanceTestHistory history = resultsStore.getTestResults(teamCityExecutionsOfSameScenario.get(0).getPerformanceExperiment(), DEFAULT_RETRY_COUNT, PERFORMANCE_DATE_RETRIEVE_DAYS, ResultsStoreHelper.determineChannelPatterns(), teamcityBuildIds);
```

Comment:
Any idea why it sometimes worked before when the channel hasn't been written here as well?

Reply:
Nova idea... I debugged into the code and found this one.